File: src/roco_serverchan_notifier/push_provider_senders/dingtalk_bot.py
from __future__ import annotations

from ..push_http import HttpSession, JsonPostRequest, post_json
from ..push_models import NotificationMessage, ProviderConfig, PushResult
from ..push_provider_auth import append_dingtalk_sign


def send_dingtalk_bot(
    provider: ProviderConfig, message: NotificationMessage, session: HttpSession, timeout: int
) -> PushResult:
    webhook = append_dingtalk_sign(provider.config["webhook"], str(provider.config.get("secret") or ""))
    payload = {
        "msgtype": "markdown",
        "markdown": {"title": message.title, "text": message.markdown},
    }
    return post_json(JsonPostRequest(provider, session, webhook, payload, timeout))

File: cloudflare-worker/src/push-provider-senders/dingtalk_bot.ts
import type { NotificationMessage, ProviderConfig, PushResult } from "../types";
import { postJson } from "../push-http";
import { appendDingTalkSign } from "../push-provider-auth";

export async function sendDingTalkBot(
  provider: ProviderConfig,
  message: NotificationMessage,
  timeoutSec: number
): Promise<PushResult> {
  const webhook = await appendDingTalkSign(
    provider.config.webhook,
    provider.config.secret || ""
  );
  const payload = {
    msgtype: "markdown",
    markdown: { title: message.title, text: message.markdown },
  };
  return postJson(provider, webhook, payload, timeoutSec);
}
